enhancements/local-storage/health-monitoring.md

```diff
@@ -0,0 +1,155 @@
+
+After that, the maintainance effort will mostly centered on the code the image runs to get health and update its `.prom` file. The logic for this collector can be in a subdirectory of CMO and have it's own OWNERS file allowing the ODF team to own it.
+
+We will be collecting metrics via [libstoragemgmt](https://github.com/libstorage/libstoragemgmt) and calling it via [libstoragemgmt-golang](https://github.com/libstorage/libstoragemgmt-golang/).
+If a block device is a partion or logical volume, we will only expose health for the parent device.
+
+
+### Ownership and Maintenance
+
+- The `node_exporter` daemonset creation is owned by the Openshift-Monitoring team. The PR for modifying the daemonset based on infrastructure to include the sidecar can be contributed by ODF, but Monitoring team will have to review.
+- The collector sidecar image and logic will live in a separate repository owned and maintained by the ODF team.
+- A new bugzilla subcomponent will be created under openshift-monitoring, but the default assignee will be ODF and ODF will be responsible for triaging the issues.
+
+### Risks and Mitigations
+
+- The golang bindings for libstoragemgmgt need some review and polish.
+  - Mitigation: The ODF(OpenShift Data Foundation) team will contribute to and help the RHEL team maintain libstoragemgmt golang
+
+## Design Details
+
+Each device will have a metric `storage_device_health` metric associated with it that has a value of 0,1, or 2. Where 0 is healthy state, 1 is a warning state, and 2 is a failed state. Each metric will be labeled by node, device-name, and device-id (where available).
+
+| storage_device_health value | Meaning |
+| --------------------------- | ------- |
+| 0                           | Healthy |
+| 1                           | Warning |
+| 2                           | Failed  |
+
+### Open Questions
+
+### Test Plan
+
+**Note:** *Section not required until targeted at a release.*
+
+- E2E tests will deploy the component and verify that metrics are being emitted per device
+
+### Graduation Criteria
+
+**Note:** *Section not required until targeted at a release.*
+
+**For non-optional features moving to GA, the graduation criteria must include
+end to end tests.**
+
+#### Dev Preview -> Tech Preview
+
+- Ability to utilize the enhancement end to end
+- End user documentation, relative API stability
+- Sufficient test coverage
+- Gather feedback from users rather than just developers
+- Write symptoms-based alerts for the component(s)
+
+#### Tech Preview -> GA
+
+- More testing (upgrade, downgrade, scale)
+- Sufficient time for feedback
+
+#### Removing a deprecated feature
+
+Does not apply
+
+### Upgrade / Downgrade Strategy
+
+
+
+### Version Skew Strategy
+
+
+
+## Implementation History
+
+
+## Drawbacks
+
+Because writing files to be read by the textfile collector cannot be done on-demand in response to a request from prometheus, we will have to poll at our own frequency.
```

Comment:
We could also have the future exporter behave as a regular exporter with a /metrics endpoint.